corehq/apps/geospatial/tasks.py

```diff
@@ -14,3 +28,42 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@task(queue='geospatial_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, INDEX_ES_TASK_HELPER_BASE_KEY)
+    if celery_task_tracker.is_active():
+        return
+
+    geo_case_prop = get_geo_case_property(domain)
+    query = case_query_for_missing_geopoint_val(domain, geo_case_prop)
+    doc_count = query.count()
+    if doc_count > MAX_GEOSPATIAL_INDEX_DOC_LIMIT:
+        celery_task_tracker.mark_as_error(error_slug='TOO_MANY_CASES')
+        return
+
+    celery_task_tracker.mark_requested()
+    batch_count = get_batch_count(doc_count, DEFAULT_QUERY_LIMIT)
+    try:
+        for i in range(batch_count):
+            docs_left = doc_count - (DEFAULT_QUERY_LIMIT * i)
+            limit = min(DEFAULT_QUERY_LIMIT, docs_left)
+            process_batch(
+                domain,
+                geo_case_prop,
+                case_type=None,
+                query_limit=limit,
+                chunk_size=DEFAULT_CHUNK_SIZE,
+                offset=i * DEFAULT_QUERY_LIMIT,
+            )
+            celery_task_tracker.update_progress(current=i + 1, total=batch_count)
```

Comment:
Cool, we are using batches to show progress.

Just suggesting an alternate way to simply multiply batch number/count by chunk size to get number of cases, which could be more user friendly.

Reply:
As an interesting thought exercise, I tried working out the results of both methods to see what they would look like and came up with the following example:

```
Assuming the following:
- current_batch = 2
- total_batches = 8
- chunk_size = 100
- total_cases = 715

Using batches:
current_batch / total_batches = 2 / 8 = 25%

Using cases:
current_batch * chunk_size / total_cases = 2 * 100 / 715 = 27.9%
```
It doesn't look like the result from both methods varies widly, but I do agree that using case count can give a slightly more accurate representation of progress. It's a very straightforward change to make, so I'll update the progress tracking to use case count instead.

Reply:
Addressed in [61e8e5b](https://github.com/dimagi/commcare-hq/pull/35126/commits/61e8e5b6d257df8da0b3680c7e6abf399ded3b25).